mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -0,0 +1,116 @@
+        experiment = ElasticExperiment.get(id=experiment_id)
+        return experiment
+
+    def create_run(self, experiment_id, user_id, start_time, tags):
+        run_id = uuid.uuid4().hex
+        # experiment = self.get_experiment(experiment_id)
+        # artifact_location = append_to_uri_path(experiment.artifact_location, run_id,
+        #                                       ElasticsearchStore.ARTIFACTS_FOLDER_NAME)
+        artifact_location = "artifact_path"
+
+        tags_dict = {}
+        for tag in tags:
+            tags_dict[tag.key] = tag.value
+        run_tags = [ElasticTag(key=key, value=value) for key, value in tags_dict.items()]
+        run = ElasticRun(meta={'id': run_id},
+                         experiment_id=experiment_id, user_id=user_id,
+                         status=RunStatus.to_string(RunStatus.RUNNING),
+                         start_time=start_time, end_time=None,
+                         lifecycle_stage=LifecycleStage.ACTIVE, artifact_uri=artifact_location,
+                         tags=run_tags)
+        run.save()
+        return run.to_mlflow_entity()
+
+    def get_run(self, run_id):
+        run = self._get_run(run_id=run_id)
+        return run.to_mlflow_entity()
+
+    def _get_run(self, run_id):
+        run = ElasticRun.get(id=run_id)
+        return run
+
+    def log_metric(self, run_id, metric):
+        run = self._get_run(run_id=run_id)
+        new_metric = ElasticMetric(key=metric.key,
+                                   value=metric.value,
+                                   timestamp=metric.timestamp,
+                                   step=metric.step)
+        run.metrics.append(new_metric)
+        run.save()
+
+    def log_param(self, run_id, param):
+        run = self._get_run(run_id=run_id)
+        new_param = ElasticParam(key=param.key,
+                                 value=param.value)
+        run.params.append(new_param)
+        run.save()
+
+    def set_tag(self, run_id, tag):
+        run = self._get_run(run_id=run_id)
+        new_tag = ElasticTag(key=tag.key,
+                             value=tag.value)
+        run.tags.append(new_tag)
+        run.save()
+
+    def search_runs(self, experiment_ids):
+        response = Search(index="mlflow-runs").filter("match",
+                                                      experiment_id=experiment_ids[0]).execute()
+        runs = []
+        for r in response:
+            metrics = []
+            params = []
+            tags = []
+            for m in r.metrics:
+                metrics.append(ElasticMetric(key=m.key,
+                                             value=m.value,
+                                             timestamp=m.timestamp,
+                                             step=m.step))
+            for p in r.params:
+                params.append(ElasticParam(key=p.key, value=p.value))
+            for t in r.tags:
+                tags.append(ElasticTag(key=t.key, value=t.value))
```

Comment:
Do you really need to loop over the metrics, params and tags here ? It will slow down the function quite a lot I think